maintainers/maintainer-list.nix

```
@@ -11521,6 +11521,15 @@
     githubId = 279868;
     name = "Matti Kariluoma";
   };
+  mattleon = {
+    github = "leonm1";
+    githubId = 32306579;
+    keys = [{
+      fingerprint = "C12D F14B DC9D 64E1 44C3  4D8A 755C DA4E 5923 416A";
+    }];
+    matrix = "@mattleon:matrix.org";
+    name = "Matt Leon";
+  };
```

please split this out into it's own commit called: `maintainers: add mattleon`

---

And I think
```suggestion
  leonm1 = {
    github = "leonm1";
    githubId = 32306579;
    keys = [{
      fingerprint = "C12D F14B DC9D 64E1 44C3  4D8A 755C DA4E 5923 416A";
    }];
    matrix = "@mattleon:matrix.org";
    name = "Matt Leon";
  };
```
is preferred.

---

We are not telling people what maintainer name they have to choose, so please don't.

---

https://github.com/NixOS/nixpkgs/blob/684e1e86e2eff075300c5e59b4588e80e7ab188f/maintainers/maintainer-list.nix#L34

---

But if they already decided against that we don't have to ask them again.

---

Made a separate commit to follow [pkgs/README.md#commit-conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions). I updated my maintainer handle to follow the recommendation, since I understand that can be confusing to some and don't want to cause confusion.